microsoft/testsuites/mshv/mshv_secure_boot.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import Path
+from typing import Any, Dict, cast
+
+from lisa import (
+    Logger,
+    Node,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+)
+from lisa.features.security_profile import SecureBootEnabled
+from lisa.operating_system import CBLMariner
+from lisa.sut_orchestrator import AZURE
+from lisa.testsuite import simple_requirement
+from lisa.tools import Journalctl, Reboot
+from lisa.util import (
+    SkippedException,
+    TcpConnectionException,
+    UnsupportedDistroException,
+)
+from lisa.util.constants import (
+    ENVIRONMENTS_NODES_REMOTE_ADDRESS,
+    ENVIRONMENTS_NODES_REMOTE_PORT,
+)
+from lisa.util.shell import wait_tcp_port_ready
+
+
+@TestSuiteMetadata(
+    area="mshv",
+    category="functional",
+    description="""This test suite covers the secure boot flow for
+    Dom0 AzureLinux nodes.
+    """,
+)
+class Dom0SecureBootTestSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        if not isinstance(node.os, CBLMariner):
+            raise SkippedException(
+                UnsupportedDistroException(
+                    node.os, "Dom0 secure boot test supports only Azure Linux."
+                )
+            )
+
+    @TestCaseMetadata(
+        description="""This test verifies secure boot succeeds
+        on an Azure Linux Dom0 node.
+
+        Steps:
+        1. On first boot, install the Dom0 components
+            (e.g. kernel-mshv, mshv, mshv-bootloader-lx, hvloader)
+        2. Reboot the VM (The Dom0 components should be loaded)
+        3. Await the VM to be ready by checking TCP port connectivity
+        4. Verify that secure boot is enabled by checking
+            journalctl for "Secure boot enabled"
+        5. Verify that the Dom0 stack is running by checking
+            journalctl for "Hyper-V: running as root partition"
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_features=[SecureBootEnabled()],
+            supported_platform_type=[AZURE],
+        ),
+    )
+    def verify_mshv_secure_boot_succeeds(
+        self,
+        log: Logger,
+        node: RemoteNode,
+        log_path: Path,
+        variables: Dict[str, Any],
+    ) -> None:
+        os: CBLMariner = cast(CBLMariner, node.os)
+
+        # 1. Install the Dom0 components
+        # Can provide: mshv, mshv-bootloader-lx
+        source_mshv_tarball = variables.get("mshv_rpm_tarball")
+        if source_mshv_tarball:
+            os.add_repository(source_mshv_tarball)
+
+        # Can provide: kernel-mshv, hvloader
+        source_base_tarball = variables.get("base_rpm_tarball")
+        if source_base_tarball:
+            os.add_repository(source_base_tarball)
+
+        components = ["kernel-mshv", "mshv", "mshv-bootloader-lx", "hvloader"]
+        os.install_packages(components)
```

Comment:
If user doesn't provide source_mshv_tarball and source_base_tarball, what will happen here?